client/extensions/woocommerce/state/sites/products/reducer.js

```diff
@@ -139,3 +143,9 @@ function setLoading( state, page, newStatus ) {
 	isLoading[ page ] = newStatus;
 	return isLoading;
 }
+
+function setError( state, page, newStatus ) {
```

Comment:
I'll be honest, I'm not a huge fan of holding error state for each page load individually like this, but I can't think of a simpler way to do this better right now. I think it's okay to do it this way for now, but perhaps we should be thinking of a way to improve/consolidate code like this in the future.